orchagent/portsorch.cpp

```diff
@@ -2172,7 +2172,13 @@ bool PortsOrch::createVlanHostIntf(Port& vl, string hostif_name)
     attrs.push_back(attr);
 
     attr.id = SAI_HOSTIF_ATTR_NAME;
-    strncpy(attr.value.chardata, hostif_name.c_str(), sizeof(attr.value.chardata));
+    if (hostif_name.length() >= SAI_HOSTIF_NAME_SIZE)
```

Comment:
@stephenxs I wonder why not to raise a warning and still set it. it indeed will not be as the user requested but much closer that it was before.

Reply:
Hi @liat-grozovik 
According to the SAI interface, the maximum bytes in `SAI_HOSTIF_ATTR_NAME` is `SAI_HOSTIF_NAME_SIZE - 1`. So I'm afraid setting it to a string longer than the size can incur SAI error which eventually aborts orchagent.
```
    /**
     * @brief Name [char[SAI_HOSTIF_NAME_SIZE]]
     *
     * The maximum number of characters for the name is SAI_HOSTIF_NAME_SIZE - 1 since
     * it needs the terminating null byte ('\0') at the end.
     *
     * If Hostif is a generic netlink, this indicates the generic netlink family name.
     *
     * @type char
     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
     * @condition SAI_HOSTIF_ATTR_TYPE == SAI_HOSTIF_TYPE_NETDEV or SAI_HOSTIF_ATTR_TYPE == SAI_HOSTIF_TYPE_GENETLINK
     */
    SAI_HOSTIF_ATTR_NAME,
```
Meanwhile, the maximum size of net_device ifname is also 15 bytes. So this is just a protection and to eliminate compiling warnings. In most cases, it won't introduce a different behavior as it did.

Reply:
> @stephenxs I wonder why not to raise a warning and still set it. it indeed will not be as the user requested but much closer that it was before.

Updated accordingly.